src/psyclone/domain/common/algorithm/psyir.py

```diff
@@ -271,6 +278,31 @@ def lower_to_language_level(self):
         call = Call.create(routine_symbol, arguments)
         self.replace_with(call)
 
+        # Remove original invoke symbol if there are no other
+        # references to it. This is not strictly necessary but it
```

Comment:
I think it is necessary - code generation doesn't work without it?

Reply:
I've modified the words. My point was that it is possible to have symbols in the symbol table that are not referenced, the actual reason for an exception is that this symbol has an unresolved interface (and in fact our previous solution was to simply change the type of interface for this symbol).